### PR TITLE
WebXRManager: Removed getCamera() and added setUserCamera().

### PR DESCRIPTION
**Pull request by mrdoob**
**Description**

If we update the transform of the app's camera inside `WebXRManager` we end up with a rotated camera when leaving XR.

I find it nicer to be able to use `OrbitControls` in the app and not have affected by going in/out of XR.

However, in case someone still needs the previous behaviour then can use `xr.setUserCamera()`.

### CODE REVIEW

File: src/renderers/webxr/WebXRManager.js [hunk 1]
```diff
@@ -55,20 +57,28 @@ class WebXRManager extends EventDispatcher {
...
 
 		//
 
-		this.cameraAutoUpdate = true;
+		this.cameraAutoUpdate = true; // @deprecated, r153
 		this.enabled = false;
 
 		this.isPresenting = false;
 
+		this.getCamera = function () {}; // @deprecated, r153
```
Comment (LeviPesin) [inline src/renderers/webxr/WebXRManager.js:74]: Maybe we should log a warning?

**Comment by jrjdavidson** [comment]
i use `xr.getCamera()` to set layers on the VR cameras like this : 
```
	const VRCamera = renderer.xr.getCamera();
	VRCamera.layers.enable( 3 );
	VRCamera.layers.disable( 0 );
	VRCamera.cameras.forEach( camera => {

		camera.layers.enable( 3 );
		camera.layers.disable( 0 );

	} );
```

I use it this way as I have a spherical mesh on layer 3 that i display a video on, and switch between the video and the rest of the scene by switching the camera layer. I find it easier this was than using the `visible` property on each every mesh on the scene. Is this a valid use of `xr.getCamera()`?

**Comment by ashconnell** [comment]
@mrdoob would you re-consider a PR to add `getCamera()` back in?

We use this to configure layers on the cameras in such a way that `setUserCamera()` does not allow:


```jsx
configureXRCamera() {
    /**
     * HACK
     * ----
     * The WebXRManager uses its own cameras and they don't match the layers
     * configured on our main scene camera.
     * This is worsened by the fact that we don't have an easy way to preconfigure
     * these camera layers as WebXRManager.getCamera() returns an empty ArrayCamera
     * that isn't populated until XR starts.
     * In addition, for some reason both the ArrayCamera and its sub-cameras need to
     * have layers all updated.
     */
    if (this.xrCameraConfigured) return
    if (this.xrCamera.cameras.length) {
      // configure main xr camera
      this.xrCamera.layers.enableAll()
      this.xrCamera.layers.disable(GraphicsLayers.MIRROR)
      // configure left eye camera
      this.xrCamera.cameras[0].layers.enableAll()
      this.xrCamera.cameras[0].layers.disable(GraphicsLayers.XR_RIGHT_EYE)
      this.xrCamera.cameras[0].layers.disable(GraphicsLayers.MIRROR)
      // configure right eye camera
      this.xrCamera.cameras[1].layers.enableAll()
      this.xrCamera.cameras[1].layers.disable(GraphicsLayers.XR_LEFT_EYE)
      this.xrCamera.cameras[1].layers.disable(GraphicsLayers.MIRROR)
      // never run this again!
      this.xrCameraConfigured = true
    }
  }
```

**Comment by marwie** [comment]
We have the same problem here in Needle Engine where we need access to the xrCamera / array camera.

**Comment by hybridherbst** [comment]
@mrdoob If three doesn't sync the XR camera to the current main camera under the hoods anymore, ALL code that does something with the camera now needs to be aware of the existence of XR and specifically handle it.

This breaks things such as billboards/look at, spatial audio, potentially raycasting if people are not careful, etc etc. - it makes "XR support out of the box" much harder. Is that intended?

**Comment by mrdoob** [comment]
Yeah... I'm going to revert this.
I ended up not having enough time to refactor all this.

**Comment by marwie** [comment]
Thank you!

**Comment by hybridherbst** [comment]
Thank you! I think it's still worth thinking about the case - I agree that it is somewhat of an issue (as seen with OrbitControls) that the main camera can't really be independently handled while in VR.

**Pull request merged**